### PR TITLE
gCloud KMS secrets backend

### PR DESCRIPTION
**Pull request by adrianchifor**
Adds support for Google Cloud KMS secrets backend. To use it, we need to first create a keyring and key on KMS, then configure it in a target or class in the inventory:
```
parameters:
  kapitan:
    secrets:
      gkms:
        key: 'projects/<project>/locations/<location>/keyRings/<keyRing>/cryptoKeys/<key>'
```
and then use it, like in our example:
```
...
  mysql:
    password: ?{gkms:mysql/password|randomstr|base64}
```
which on compile, will generate a random b64-encoded string encrypted with the key at `parameters.kapitan.secrets.gkms.key` and save it in `secrets/mysql/password`.

By default, it will use gcloud default credentials, but can be configured to use a service account by setting the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.

### CODE REVIEW

File: kapitan/refs/secrets/gpg.py [hunk 1]
```diff
@@ -158,7 +164,6 @@ def __init__(self, path, ref_type=GPGSecret):
         "init GPGBackend ref backend type"
         super().__init__(path, ref_type)
         self.type_name = 'gpg'
-        self.gpg = gpg_obj()
```
[the remark this answers is not in the document]
Comment (ramaro) [reply kapitan/refs/secrets/gpg.py]: let's remove it then

File: kapitan/cli.py [hunk 2]
```diff
@@ -171,10 +172,13 @@ def main():
     secrets_parser.add_argument('--inventory-path',
                                 default=from_dot_kapitan('secrets', 'inventory-path', './inventory'),
                                 help='set inventory path, default is "./inventory"')
-    secrets_parser.add_argument('--recipients', '-R', help='set recipients',
+    secrets_parser.add_argument('--recipients', '-R', help='set GPG recipients',
                                 type=str, nargs='+',
                                 default=from_dot_kapitan('secrets', 'recipients', []),
                                 metavar='RECIPIENT')
+    secrets_parser.add_argument('--key', '-K', help='set KMS key',
```
[the remark this answers is not in the document]
Comment (adrianchifor) [reply kapitan/cli.py]: You don't, key is taken from the file. It's the same mindset as --recipients, you only need to set it when using --write/--update and you don't want to use --target.

[the remark this answers is not in the document]
Comment (ramaro) [reply kapitan/cli.py]: Got it.

File: kapitan/cli.py [hunk 3]
```diff
@@ -301,17 +305,33 @@ def secret_write(args, ref_controller):
...
         recipients = [dict((("name", name),)) for name in args.recipients]
         if args.target_name:
             inv = inventory_reclass(args.inventory_path)
-            # TODO move into kapitan:secrets:gpg:recipients key
-            recipients = inv['nodes'][args.target_name]['parameters']['kapitan']['secrets']['recipients']
+            try:
+                recipients = inv['nodes'][args.target_name]['parameters']['kapitan']['secrets']['gpg']['recipients']
+            except KeyError:
+                # TODO: Keeping gpg recipients backwards-compatible until we make a breaking release
+                logger.info("WARNING: parameters.kapitan.secrets.recipients is deprecated, " +
```
[the remark this answers is not in the document]
Comment (adrianchifor) [reply kapitan/cli.py]: good point, done

**Pull request merged**